docs/user-guide/arrays.rst

```diff
@@ -0,0 +1,487 @@
+
+Working With Arrays
+===================
+
+Creating an array
+-----------------
+
+Zarr has several functions for creating arrays. For example:
+
+.. ipython:: python
+
+   import zarr
+
+   z = zarr.zeros((10000, 10000), chunks=(1000, 1000), dtype='i4')
+   z
+
+The code above creates a 2-dimensional array of 32-bit integers with 10000 rows
+and 10000 columns, divided into chunks where each chunk has 1000 rows and 1000
+columns (and so there will be 100 chunks in total).
+
+For a complete list of array creation routines see the :mod:`zarr.api.synchronous`
+module documentation.
+
+.. _tutorial_array:
+
+Reading and writing data
+------------------------
+
+Zarr arrays support a similar interface to NumPy arrays for reading and writing
+data. For example, the entire array can be filled with a scalar value:
+
+.. ipython:: python
+
+   z[:] = 42
+
+Regions of the array can also be written to, e.g.:
+
+.. ipython:: python
+
+   import numpy as np
+
+   z[0, :] = np.arange(10000)
+   z[:, 0] = np.arange(10000)
+
+The contents of the array can be retrieved by slicing, which will load the
+requested region into memory as a NumPy array, e.g.:
+
+.. ipython:: python
+
+   z[0, 0]
+   z[-1, -1]
+   z[0, :]
+   z[:, 0]
+   z[:]
+
+.. _tutorial_persist:
+
+Persistent arrays
+-----------------
+
+In the examples above, compressed data for each chunk of the array was stored in
+main memory. Zarr arrays can also be stored on a file system, enabling
+persistence of data between sessions. For example:
+
+.. ipython:: python
+
+   z1 = zarr.open('data/example.zarr', mode='w', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```

Comment:
Should we replace all uses of `zarr.open`, `zarr.array`, `zarr.group` in favor of `zarr.{create,open}_{array,group}` once #2463 lands?